### PR TITLE
Use 'main' as the default branch

### PR DESCRIPTION
**Pull request by david-caro**
This might require tweaking some configuration in the repository.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
   flake8_py3:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
```
Comment (david-caro) [inline .github/workflows/lint.yaml:10]: This is not directly related to us, but also works :)

**Review by chicocvenancio — APPROVED** [review]
We should set branch protection on main as well. Will do.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (chicocvenancio) [reply .github/workflows/lint.yaml:10]: do we want to pin this?

**Comment by david-caro** [comment]
> We should set branch protection on main as well. Will do.

I just found that you can just rename the `master` branch and will copy over the permissions, and rebase the existing PRs, maybe that's better?

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (david-caro) [reply .github/workflows/lint.yaml:10]: Not sure, it wasn´t before, so I kept it like it was, do you foresee any issues? (if so I'll pin it in a different PR)

Comment (chicocvenancio) [reply .github/workflows/lint.yaml:10]: @david-caro I mean, it is unlikely since we use it without arguments and its probably the most used action from github itself. But in theory a new version could change behaviour and break something in our CI. 

CC @supertassu

Comment (david-caro) [reply .github/workflows/lint.yaml:10]: Makes sense yes, I'll open a PR for it :)
